Analysis/src/EventTags.cpp

```diff
@@ -48,49 +59,45 @@ EventTags EventTagCreator::CreateEventTags(const std::vector<DataId>& dataIds_ba
+
+    //boost::optional<DiscriminatorWP> vbf_tag;
+    //if(vbf_tag_raw > 0) vbf_tag = static_cast<DiscriminatorWP>(vbf_tag_raw);
 
     EventTags evt_tags;
 
     for (size_t i = 0; i < dataIds_base.size(); ++i) {
         const auto& dataId_base = dataIds_base.at(i);
-        const double weight = weights.at(i);
+        double weight = weights.at(i);
 
         if(dataId_base.Get<EventCategory>() != evtCategory_2j
                 || !event_unc_sources.count(dataId_base.Get<UncertaintySource>()))
             continue;
 
         for(const auto& category : categories) {
-            if(!category.Contains(static_cast<size_t>(num_central_jets), bjet_counts, is_vbf, is_boosted, vbf_tag))
+            if(!category.Contains(static_cast<size_t>(num_central_jets), bjet_counts, is_vbf, is_boosted, vbf_cat.second))
                 continue;
             EventSubCategory evtSubCategory;
-            if(has_b_pair) {
-                if(category.HasBoostConstraint() && category.IsBoosted()) {
-                    if(use_svFit) {
-                        const bool isInsideBoostedCut = cuts::hh_bbtautau_Run2::hh_tag::IsInsideBoostedMassWindow(SVfit_p4.M(), m_bb);
-                        evtSubCategory.SetCutResult(SelectionCut::mh, isInsideBoostedCut);
-                    }
-                } else {
-                    if(!use_svFit && massWindowParams.count(SelectionCut::mh))
-                        throw exception("Category mh inconsistent with the false requirement of SVfit.");
-                    if(massWindowParams.count(SelectionCut::mh)) {
-                        const bool cut_result = use_svFit
-                                    //&& event.GetSVFitResults(ana_setup.allow_calc_svFit).has_valid_momentum
-                                    && massWindowParams.at(SelectionCut::mh).IsInside(
-                                    SVfit_p4.M(), m_bb);
-                        evtSubCategory.SetCutResult(SelectionCut::mh, cut_result);
-                    }
-                    if(massWindowParams.count(SelectionCut::mhVis))
-                        evtSubCategory.SetCutResult(SelectionCut::mhVis,massWindowParams.at(SelectionCut::mhVis)
-                                .IsInside(m_tt_vis, m_bb));
-
-                    if(massWindowParams.count(SelectionCut::mhMET))
-                        evtSubCategory.SetCutResult(SelectionCut::mhMET,massWindowParams.at(SelectionCut::mhMET)
-                                .IsInside((SVfit_p4+MET_p4).M(), m_bb));
+            double btag_sf = 1;
+            if(category.HasBtagConstraint() && !btag_weights.empty()){
+                  btag_sf = btag_weights.at(category.BtagWP());
+            }
+            weight *= btag_sf;
 
-                }
-                if(use_kinFit)
+            if(has_b_pair) {
+              const EllipseParameters& window = category.HasBoostConstraint() && category.IsBoosted() ? cuts::hh_bbtautau_Run2::hh_tag::boosted_window : cuts::hh_bbtautau_Run2::hh_tag::resolved_window;
```

Comment:
Have you changed definitions of mass windows in the header? If yes, you also need to do a PR in h-tautau repository.